detect_secrets/plugins/common/yaml_file_parser.py

```diff
@@ -38,10 +38,11 @@ class YamlFileParser(object):
     This parsing method is inspired by https://stackoverflow.com/a/13319530.
     """
 
-    def __init__(self, file):
+    def __init__(self, file, exclude_lines_re=None):
```

Comment:
Please specify the type you expect for this. It seems like a regex object, but that can get confusing since we pass in a string in `BasePlugin.__init__`.